# Assignment 1: Data Visualization and Description (MAT205)
## Weightage: 5%
## Due: 30th September, 2022
## Directions:

   - You are required to do all the exercises in this jupyter notebook and submit it through VLE Assignment submission section
   - Plagiarism will be taken seriously, so make sure what you submit is your own work
   - Late submissions will be penalized as per the Wheel of Academic Law (WAL)
   - Under each Question, an empty cell is kept so that you can provide your answers. If you need more cells for your answers, just add few more cells below and continue.
   - It is upto the individuals to change the colour of the plots for visual appeal, BUT do not overdo it.
   - For each plot, give the appropriate title and label the axes accordingly. 
   - While submitting your work, rename the file as your stdnumber_class.ipynb, for example, 02210062_2IT.ipynb

### Insert your Student Number and Class below:

### Student Number: 
### Class:

## Introduction

These exercises are based on the popular dataset, the Sinking of the Titanic in the year 1911. The first part of the exercises will be based on descriptive statistics and data visualization using the matplotlib library, whereas the second half will emphasize on the visualization using the seaborn library.

## Part I: Data Visualization using the Matplotlib library

#### Import the necessary libraries as follows:

In [5]:
import numpy as np
import pandas as pd  # To import data as DataFrame. Pandas is built on NumPy so you need that as well.
import matplotlib.pyplot as plt # For plotting; visualization in Part I.
import seaborn as sns # For visualization in part II

#### In this section you will be using only the plt function to obtain all the necessary plots. 
#### Use the plt.show() function to display your plots!

#### Now we need to load the titanic dataset, which comes with the seaborn library. However, you will need an INTERNET CONNECTION to load this dataset!!!

In [20]:
sns.get_dataset_names()   # Available datasets with Seaborn but we will only use the titanic dataset.
# If you are interested, you may explore other datasets as well!

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [7]:
df = sns.load_dataset("titanic") 
# Dataset is now assigned to a variable df. So we refer to the dataset as df from hereon!

In [9]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [10]:
df.shape

(891, 15)

- As we it is observed, this dataset has 891 rows and 15 columns. The datatypes are different: int64, float64, object, categorical, etc. To get the information on the datatypes and number of elements, you can use the info() function as follows:

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


## Questions:
### 1. Obtain the descriptive statistics summary of the given dataframe using the function describe().

### Answer here

### 2. Plot Histograms of the passengers on board with their:

(i) Age \
(ii) Fare paid

### 3. Plot the histogram of the Age of the passengers by their sex:

(i) Stacked histogram \
(ii) Side by side histogram

### 4. Display the Box and Whisker plot of the fare paid by the passengers.

### 5. Obtain the Box and Whisker plot with age of the passengers.
- Here, since the age column contains some non-numerical elements, we have to obtain the rows without NaN entries using the following line of code.
- You will then use the new dataframe to obtain the desired plot.

In [15]:
df2 = df[df['age'].notna()]

### 6. Show the Box and Whisker plot of the age of the passengers, side by side based on the condition, whether:
(i) the passenger survived\
(ii) the passenger was alone \
(iii) the passenger was male or female

### 7. Show the Box and Whisker plot with the fare of the passengers, side by side based on the condition, whether:
(i) the passenger survived\
(ii) the passenger was alone \
(iii) the passenger was male or female

### 8. Plot the line chart of the following:
(i) age of the passengers\
(ii) fare paid by the passengers

### 9. Create a scatter plot presenting the relationship between the age and the fare paid by the passengers.

### 10. How many passengers survived the disaster?

### 11. How many passengers embarked from Southhampton?

### 12. Create the Bar plot of the following:
   #### (i) passengers with following age ranges:
- less than 15
- between 15 and 30
- between 30 and 45
- between 45 and 60
- above 60 years of age

#### (ii) passengers in the First, Second and Third class
#### (iii) passengers who embarked from Southampton, Cherbourg, and Queenstown.
  

## Part II: Data Visualization using the Seaborn library

#### For plots in this part, you will use the sns function.

### 1. Creat the Histogram on the following:
#### (i) Age of the passengers
#### (ii) Fare paid by the passengers

### 2. Create a stacked Histogram of the age of passengers differing by whether they survived or not? Is this a good way to present the information for comparison in the dataset?

### 3. Create the Histograms on the following. Histograms in each case must be side by side:
### (i) Age of the passengers differing by 
   ##### (a) whether they survived or not
   ##### (b) sex
   ##### (c) whether they were alone
### (ii) Fare paid by the passengers differing by 
   ##### (a) whether they survived or not
   ##### (b) sex
   ##### (c) whether they were alone

### 4. Plot the Histogram of passengers who embarked from Cherbourg, differing by their survival status from the accident. Display the plots side by side.

### 5. Plot the Kernel Density Estimate or the Ogive graph on the age of the passengers. 

### 6. Plot the ogive curve on the age of the passengers differing by:
#### (i) survival 
#### (ii) class
#### (iii) whether they were alone or not

### 6. Draw the cumulative frequency distribution curve of the passengers on the following:
#### (i) Age
#### (ii) Fare 

### 7. Plot the Ogive graph on top of the histogram on the age of the passengers.

### 8. Make the Box and Whisker plots on the following:
#### (i) Age of the passengers differing by their survival
#### (ii) Age of the passengers differing by their class
#### (ii) Fare paid by the passengers differing by their survival

### 9. Create a scatter plot presenting the relationship between the age and fare paid by the passengers.

### 10. Create a scatter plot of the Fare paid and the Age. Differ the plot colour by 
#### (i) gender, 
#### (ii) survival, and 
#### (iii) whether they were alone or not
#### (iv) embarking place

## BONUS!!!
### Create a few questions on your own on the given dataset and answer it.